scipy/fft/_pocketfft/setup.py

```diff
@@ -0,0 +1,76 @@
+
+# TODO: As of python 3.6, CCompiler has a `has_flag` method to use instead.
+def has_flag(compiler, flagname):
+    # Return True if the flag is supported by the compiler
+    import tempfile
+    from distutils.errors import CompileError
+    with tempfile.NamedTemporaryFile('w', suffix='.cpp') as f:
+        f.write('int main (int argc, char **argv) { return 0; }')
+        try:
+            compiler.compile([f.name], extra_postargs=[flagname])
+        except CompileError:
+            return False
+    return True
+
+
+def get_std_flag(compiler):
+    # Test the compiler for the highest available c++ standard flag
+    gnu_flags = ['--std=c++14', '--std=c++11']
+    flags_by_cc = {
+        'msvc': ['/std:c++14', None],
+        'intelw': ['/Qstd=c++14', '/Qstd=c++11']
+    }
+    flags = flags_by_cc.get(compiler.compiler_type, gnu_flags)
+
+    for flag in flags:
+        if flag is None:
+            return None
+
+        if has_flag(compiler, flag):
+            return flag
+
+    from numpy.distutils import log
+    log.warn('Could not detect c++ standard flag')
+    return None
+
+
+def pre_build_hook(build_ext, ext):
+    cc = build_ext._cxx_compiler.compiler_type
+    args = ext.extra_compile_args
+
+    std_flag = get_std_flag(build_ext._cxx_compiler)
+    if std_flag is not None:
+        args.append(std_flag)
+
+    if cc == 'msvc':
+        args.append('/EHsc')
+    else:
+        if has_flag(build_ext._cxx_compiler, '-fvisibility=hidden'):
+            args.append('-fvisibility=hidden')
+
+        import sys
+        if sys.platform == 'darwin':
+            args += ['-stdlib=libc++', '-mmacosx-version-min=10.7']
```

Comment:
I would also use `has_flag` for `-stdlib=libc++` (some OS X users may want to build with gcc rather than clang and I don't think that will work with that flag).

 Also, I think conda-forge and anaconda currently use 10.9 for `MACOSX_DEPLOYMENT_TARGET`. Setting `-mmacosx-version-min=10.7` will override this. Is this okay to do as long as the version here is older or does setting this require the user to have an older SDK installed?

Reply:
From reading around, it seems that the flag mainly adds compiler warnings when using unavailable APIs. It doesn't sound like it changes which version of the SDK is linked.